core/src/main/java/com/amplifyframework/predictions/models/Feature.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import androidx.annotation.NonNull;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * A generic class to hold information about an inferred
+ * feature and the confidence score for inference.
+ * @param <T> the feature type
+ */
+public abstract class Feature<T> implements Comparable<Feature<T>> {
+    private final String id;
+    private final T feature;
```

Comment:
There was ONE reason why I didn't name it that way, but it's trivial enough reason that I can just worry about it later.

So there is `BoundedKeyValue` type that extends `ImageFeature` (no longer in this PR since it's for `Predictions.identify()`), and I added `getKey()` and `getValue()` methods to it to return its key and value, respectively. Now it will be forced to return the whole key-value pair upon calling `BoundedKeyValue#getValue()`.

Reply:
Ahhh, yea. Like `boundedKeyValue.getValue().getValue() // OMG, Value three times in a row?` 

Hmmm.